ocaml/libs/tgroup/tgroup.ml

```diff
@@ -0,0 +1,122 @@
+(*
+ * Copyright (C) Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT Group WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = __MODULE__ end)
+
+let ( // ) = Filename.concat
+
+let requests = "/sys/fs/cgroup/cpu/control.slice/xapi.service/request"
```

Comment:
I don't think this value should be hardcoded, the interface doesn't make it seem like it should just work for the xapi service.

Reply:
I was thinking of adding it in the unixext. Or a version of it. It's basically `pidfile_write` but the pid (in this case I want to pass the tid as a parameter) is passed as a parameter. I will also need something similar to write the cpu shares for cgroups.

Reply:
I changed `init` to accept a directory and plan to add this in `Xapi_globs`